src/main/scala/Main.scala

```diff
@@ -0,0 +1,37 @@
+import scala.annotation.tailrec
+import scala.concurrent.duration._
+
+object Main {
+
+  def main(args: Array[String]): Unit = {
+    retry[Int](block = attempt => {
+      println("block " + (attempt+1))
+      attempt+1
+    },
+      acceptResult = res => res>3,
+      retries = List(0.seconds, 1.seconds, 2.seconds, 3.second))
+  }
+
+  @tailrec
+  final def retry[A](block: Int => A,
+                     acceptResult: A => Boolean,
+                     retries: List[FiniteDuration]): A = {
+    var attempt = retries.length
+    if (attempt <= 0) {
+      println("the end")
+      throw new Exception("There is no more attempts. Method failed")
```

Comment:
exceptions are side effects, we should avoid them and in this case we can avoid them for sure :)

Reply:
also I mentioned in the class that if all retries are exceeded then the retry function should return the last value of block() function.